### PR TITLE
[CALCITE-6422] Query with '<>' throws NullPointerException during materialized view matching

### PR DESCRIPTION
**Pull request by l4wei**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java [hunk 1]
```diff
@@ -443,7 +443,8 @@ public static boolean mayBeSatisfiable(RexNode e) {
     for (RexNode disjunction : disjunctions) {
       switch (disjunction.getKind()) {
       case LITERAL:
-        if (!RexLiteral.booleanValue(disjunction)) {
+        if (!RexLiteral.isNullLiteral(disjunction)
```
[the remark this answers is not in the document]
Comment (l4wei) [reply core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java:446]: I see, the null value is generated by following method:

And the nullAs is UNKNOWN in materialized view matching.

**Review by YiwenWu — APPROVED** [review]
LGTM

**Comment by l4wei** [comment]
@YiwenWu Can you merge this PR into main branch?

**Comment by l4wei** [comment]
@YiwenWu Can you merge this PR into main branch?

**Comment by YiwenWu** [comment]
> @YiwenWu Can you merge this PR into main branch if you approve this PR?

I currently do not have merge permissions.

**Comment by mihaibudiu** [comment]
You need to squash the commits for the PR to be merged.

**Comment by mihaibudiu** [comment]
Also, your PR and your commit should use the same message as the JIRA issue title.

**Comment by l4wei** [comment]
@mihaibudiu Sure, you are right, I have followed you advices.

**Pull request merged**